### DISCUSSION TITLE
Implement new HA device based mqtt discovery

### DISCUSSION THREAD

**Discussion by KipK** [Koenkk/zigbee2mqtt#24732] [General]
HA has implemented device discovery from latest update
https://www.home-assistant.io/integrations/mqtt/?fbclid=IwY2xjawGfSQdleHRuA2FlbQIxMQABHZxVn5U6gJ0vAaSIZc7yYlEntkJnWUbMpbllGOOzHLJiMj21MHfD6MFFhA_aem_3GY8ZpzDUA7nRIBe9P6uSQ#device-discovery-payload

This is supposed to give less traffic than entity based discovery on mqtt server when Z2M starts and less topics used. Mosquito has a limit in the number of topics , with hundreds of device on the network you can reach this limit with entity based discovery. 

Interesting point, they've added a migration protocol ( explained in the doc above )  to migrate from entities to device or falling back.

**Comment by mundschenk-at** [comment]
Device discovery grew out of the discussions in https://github.com/Koenkk/zigbee2mqtt/issues/20648. It's certainly something we's like to implement, but probably not for 2.0.0. It's a pretty major overhaul and it will require everyone to update to HA 2024.11.x (unless we add a config switch, but then both code paths would have to be maintained). Also other systems may rely on the HA discovery protocol, but won't have implemented device discovery yet.

**Reply by Koenkk** [reply]
Agree with this, let's wait some time to give everyone time to upgrade.
